libnetwork/slirp4netns/slirp4netns.go

```diff
@@ -685,9 +685,13 @@ func openSlirp4netnsPort(apiSocket, proto, hostip string, hostport, guestport ui
 	if _, err := conn.Write([]byte(fmt.Sprintf("%s\n", data))); err != nil {
 		return fmt.Errorf("cannot write to control socket %s: %w", apiSocket, err)
 	}
-	if err := conn.(*net.UnixConn).CloseWrite(); err != nil {
-		return fmt.Errorf("cannot shutdown the socket %s: %w", apiSocket, err)
+	uconn, ok := conn.(*net.UnixConn)
+	if ok {
```

Comment:
(Non-blocking: I’d mildly prefer returning an error if this “can’t happen”)

Reply:
I think, it might be better to silence the linter then here. I think a panic with a sane stack trace on how we ended up here with the wrong socket is much more valuable to me then a error saying unexpected type cast.

Reply:
That works for me as well.